test/e2e/settings/global_access_block.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package settings
+
+import (
+	"net/http"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/parnurzeal/gorequest"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("Global access block", func() {
+	f := framework.NewDefaultFramework("global-access-block")
+
+	host := "global-access-block"
+
+	BeforeEach(func() {
+		err := f.NewEchoDeploymentWithReplicas(1)
+		Expect(err).NotTo(HaveOccurred())
+
+		ing, err := f.EnsureIngress(framework.NewSingleIngress(host, "/", host, f.IngressController.Namespace, "http-svc", 80, nil))
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ing).NotTo(BeNil())
+	})
+
+	AfterEach(func() {
+	})
+
+	It("should block IP addresses defined in the ConfigMap", func() {
+		err := f.UpdateNginxConfigMapData("block-ips", "172.16.0.0/12,192.168.0.0/16,10.0.0.0/8")
+		Expect(err).NotTo(HaveOccurred())
+
+		err = f.WaitForNginxConfiguration(
+			func(cfg string) bool {
+				return strings.Contains(cfg, "deny 172.16.0.0/12;") &&
+					strings.Contains(cfg, "deny 192.168.0.0/16;") &&
+					strings.Contains(cfg, "deny 10.0.0.0/8;")
+			})
+		Expect(err).NotTo(HaveOccurred())
+
+		// This test works for minikube, but may have problems with real kubernetes clusters,
+		// especially if connection is done via Internet. In this case, the test should be disabled/removed.
+		resp, _, errs := gorequest.New().
+			Get(f.IngressController.HTTPURL).
+			Set("Host", host).
+			End()
+		Expect(len(errs)).Should(BeNumerically("==", 0))
```

Comment:
This doesn't print the content of the returned error(s), making it hard to troubleshoot from my experience.
`Expect(errs).To(BeNil())` does print the content of the errors and is equally valid.

I know we have been abusing this in another test, but I it's never too late to start improving our tests :)

Reply:
Hey @antoineco , done